### PR TITLE
Amcl refactorization

### PR DESCRIPTION
**Pull request by mhpanah**
1. Created map directory and transfered map.h header file to map directory to be consistent with other libraries. 
2. Updated ament_target_dependencies for costmap to contain correct dependencies.
3. Commented out unused parameters from cpp file to eliminate compiler warnings
4. For C files had to cast unused parameters with void to eliminate warnings.
5. TODO: decide what to do with [-Wpedantic].

### CODE REVIEW

File: src/libs/nav2_util/src/pf/pf.c [hunk 1]
```diff
@@ -216,7 +216,7 @@ int pf_update_converged(pf_t *pf)
   int i;
   pf_sample_set_t *set;
   pf_sample_t *sample;
-  double total;
+  /*double total*/;
```
Comment (crdelsey) [inline src/libs/nav2_util/src/pf/pf.c]: This should be just deleted. It's not used and also not a parameter.

File: src/libs/nav2_util/src/sensors/amcl_sensor.cpp [hunk 2]
```diff
@@ -45,23 +45,23 @@ AMCLSensor::~AMCLSensor()
 
 ////////////////////////////////////////////////////////////////////////////////
 // Apply the action model
-bool AMCLSensor::UpdateAction(pf_t *pf, AMCLSensorData *data)
+bool AMCLSensor::UpdateAction(pf_t * /*pf*/, AMCLSensorData * /*data*/)
```
Comment (crdelsey) [inline src/libs/nav2_util/src/sensors/amcl_sensor.cpp]: Why do we have all these functions that don't do anything? Is this really an abstract base class? Could we make these pure virtual instead?

File: src/libs/nav2_util/src/pf/pf.c (hunk 1 above)
Comment (mhpanah) [reply src/libs/nav2_util/src/pf/pf.c]: Agreed.

File: src/libs/nav2_util/src/sensors/amcl_sensor.cpp (hunk 2 above)
Comment (mhpanah) [reply src/libs/nav2_util/src/sensors/amcl_sensor.cpp]: At this stage I don't want to change the code.  Once, I start changing the code we can address this issue.

File: src/libs/nav2_util/src/map/map.c [hunk 3]
```diff
@@ -69,6 +69,7 @@ void map_free(map_t *map)
 // Get the cell at the given point
 map_cell_t *map_get_cell(map_t *map, double ox, double oy, double oa)
 {
+  (void)oa;
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply src/libs/nav2_util/src/map/map.c]: I was trying to not change the code for this commit and start modifying the code in next phase by removing all the unnecessary parameters, changing functions, classes, etc.

File: src/libs/nav2_util/src/pf/pf.c [hunk 4]
```diff
@@ -472,6 +471,7 @@ int pf_resample_limit(pf_t *pf, int k)
 // Re-compute the cluster statistics for a sample set
 void pf_cluster_stats(pf_t *pf, pf_sample_set_t *set)
 {
+  (void)pf;
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply src/libs/nav2_util/src/pf/pf.c]: same as above.

File: src/libs/nav2_util/src/pf/pf_kdtree.c [hunk 5]
```diff
@@ -185,6 +185,7 @@ int pf_kdtree_get_cluster(pf_kdtree_t *self, pf_vector_t pose)
 // Compare keys to see if they are equal
 int pf_kdtree_equal(pf_kdtree_t *self, int key_a[], int key_b[])
 {
+  (void)self;
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply src/libs/nav2_util/src/pf/pf_kdtree.c]: same as above.

**Comment by mhpanah** [comment]
@mkhansen-intel - Agreed. I need to create an issue to de-boostify the code to meet ROS2 Developer guideline. That was my plan.

**Comment by mkhansen-intel** [comment]
@mhpanah - the last change looks good to me. Are there any other pending changes? If not and you have issues open for remaining refactors & clean ups, I think we can merge this.

**Comment by mhpanah** [comment]
I've created an issue to eliminate the usage of boost library from the code.

**Pull request merged**